### PR TITLE
Add Manual Testing to DG

### LINKED ISSUE #449: Appendix: Instructions for Manual Testing

### PR DESCRIPTION
**Pull request by B-enguin**
Add appendix for manual testing of the HomeBoss application to the DG for the current commands in the application.

Closes #449.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
...
+   5. Test Case: `customer delete a`{.swift}.</br>
+      Expected: Similar to previous.
+
+   6. Test Case: `customer delete 2`{.swift}.</br>
+      Expected: No Customer is deleted, an Error indicating invalid Customer ID is
+      shown in the feedback message.
+
+### Add Delivery
+
+1. Adding a Delivery to the application
```
Comment (jianyangg) [inline docs/DeveloperGuide.md]: Do we want to standardise if there's a full stop here?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
...
+      Expected: No new Customer is added, Error indicating that the Customer already exists is
+      shown in the feedback message.
+
+   7. Test Case: `customer add --name Gabriel --phone 987654321 --email Gabrielrocks --address RVRC Block B`{.swift}.</br>
+      Expected: No new Customer is added, Error indicating email constraints is
+      shown in the feedback message.
+
+   8. Test Case: `customer add --name Gabriel --phone 987654321 --email Gabrielrocks@gmail.com`{.swift}.</br>
+      Expected: No new Customer is added, Error indicating invalid command format is
+      shown in the feedback message.
```
Comment (jianyangg) [inline docs/DeveloperGuide.md]: Is the comma used accidentally instead of a fullstop?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
...
+   4. Test Case: `customer view -1`{.swift}.</br>
+      Expected: No new Customer details are shown, Error indicating invalid command format is
+      shown in the feedback message.
+
+   5. Test Case: `customer view a`{.swift}.</br>
+      Expected: Similar to previous.
+
+   6. Test Case: `customer view 2`{.swift}.</br>
+      Expected: No new Customer details are shown, Error indicating invalid customer ID is
+      shown in the feedback message.
```
Comment (jianyangg) [inline docs/DeveloperGuide.md]: Same as above

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
...
 
-### Deleting a customer
+2. Subsequent launches
 
-1. Deleting a customer while all customers are being shown
+   1. Relaunch the application by running `HomeBoss.jar`{.swift}.
 
-    1. Prerequisites: List all customers using the `list` command. Multiple customers in the list.
+   2. Using the `login`{.swift} command, log in into HomeBoss with the same user details entered earlier.
+      Expected: User is able to log in successfully and see the HomeBoss homepage.
```
Comment (zhonghan721) [inline docs/DeveloperGuide.md]: (This comment might be irrelevant depending on how MarkBind renders it)
Perhaps can standardise the sub-points to use 1. 2. etc to make it consistent with the Initial Launch portion

Screenshot 2023-11-09 at 10 27 43 PM

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
...
+   7. Test Case: There already exists a user account stored in the application.</br>
+      Expected: No new user is registered. Error indicating existing account is
+      shown in the feedback message.
+
+### Login
+
+1. Login to an account
+
+   1. Prerequisites: There exists a stored user in the application, with the username `Gabriel`{.swift} and
+      password `GabrielIsGreat`{.swift}
```
Comment (zhonghan721) [inline docs/DeveloperGuide.md]: Perhaps can add in the condition that the user should be logged out as it causes different outcome for the first test case.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
...
+
+   3. Test Case: `logout extra`{.swift} or other extra arguments.</br>
+      Expected: Similar to previous.
+
+### Recover Account
+
+1. Recovering user account
+
+   1. Prerequisites: Logged-in into the application with the `login`{.swift} command, and the answer of the currently
+   logged-in user's secret question is "Koko".
```
Comment (zhonghan721) [inline docs/DeveloperGuide.md]: Would it be better to leave out the logged-in state, as recover account should ideally be performed when logged out?

**Review by zhonghan721 — COMMENTED** [review]
Some general comments

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (B-enguin) [reply docs/DeveloperGuide.md]: image
Mark Bind renders it this way, would it be better to change it to a different format?

Comment (zhonghan721) [reply docs/DeveloperGuide.md]: Ahh i see, this works!

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -1853,44 +1853,698 @@ testers are expected to do more _exploratory_ testing.
...
+      is shown in the feedback message.
+
+   10. Test Case: `delivery list --date 2023-13-04`{.swift}.</br>
+      Expected: No Deliveries are listed. An Error indicating expected delivery date constraints
+      is shown in the feedback message.
+
+   11. Test Case: `delivery list --sort random`{.swift}.</br>
+       Expected: No Deliveries are listed. An Error indicating sort constraints
+       is shown in the feedback message.
+
```
Comment (juliusgambe) [inline docs/DeveloperGuide.md:2394]: Should we test for combination of `delivery list`. For example `delivery list --sort asc --date today --status CREATED`

**Review by juliusgambe — COMMENTED** [review]
One general comment for `DeliveryList`

**Review by juliusgambe — APPROVED** [review]
LGTM

**Pull request merged**